src/main/java/uk/gov/ons/ctp/response/collection/exercise/client/impl/SurveySvcRestClientImpl.java

```diff
@@ -176,4 +179,16 @@ public SurveyDTO findSurveyByRef(final String surveyRef) {
     }
     return survey;
   }
+
+  public SurveyDTO getSurveyForCollectionExercise(final CollectionExercise collex)
+      throws CTPException {
+    final SurveyDTO survey = findSurvey(collex.getSurveyId());
+
+    if (survey == null) {
+      throw new CTPException(
+          Fault.SYSTEM_ERROR, String.format("Could not find survey id %s", collex.getId()));
```

Comment:
Wouldn't RESOURCE_NOT_FOUND be a more appropriate error to throw here?

Reply:
The reason for throwing a SYSTEM_ERROR is because a survey should always exist for a collection exercise. Also looks like these CTPExceptions are translated to responses so someone updating an event should only get a 404 if the event doesn't exist not when the survey doesn't...

Reply:
fair enough